### ISSUE TITLE
risk_free_rate in max_sharpe() can't be setup

### ISSUE THREAD

**Issue by Meng-QQ** [robertmartin8/PyPortfolioOpt#238]
**Describe the bug**
The risk_free_rate in max_sharpe() always equal to its default value as 0.02 no matter the parameter value I input, which causes sharpe ratio calculated wrong.

**Expected behavior**
If risk_free_rate=0, Expected annual return=14.6%, Annual volatility = 10.6% , Sharpe Ratio shall be 14.6% / 10.6% =1.377. But ef.max_sharpe() returns sharpe ratio as 1.19 which is calculated as (14.6%-2%)/10.6%.

If risk_free_rate=0.05, Expected annual return=19.4%, Annual volatility = 14.8% , Sharpe Ratio shall be (19.4%-5%) / 14.8% =0.973. But ef.max_sharpe() returns sharpe ratio as 1.18 which is calculated as (19.4%-2%)/14.8%.

**Screenshots**
![image](https://user-images.githubusercontent.com/71325068/103166419-cb4dd980-485c-11eb-9aa1-e12029392c89.png)

**Operating system and python version:**
Win 10, python 3.7.3

**Comment by robertmartin8** [comment]
Hi @Meng-QQ,

Thanks for raising this! The problem is not in `max_sharpe`, but in `ef.portfolio_performance()` ([this line](https://github.com/robertmartin8/PyPortfolioOpt/blob/795bf9d2d4a636e662e829f7260f4640d31e4b53/pypfopt/efficient_frontier.py#L359)). 

`max_sharpe` is indeed using the correct `risk_free_rate`, but `ef.portfolio_performance` redefines a local `risk_free_rate` variable. So to get the right behaviour, you would need to do:

```python
ef.max_sharpe(risk_free_rate=0.05)
ef.portfolio_performance(verbose=True, risk_free_rate=0.05)
```

This is terrible design on my part. I will push a fix that makes `risk_free_rate` a field of the `EfficientFrontier` object (or something like that), so that it has more sensible default behaviour.

Cheers,
Robert

**Comment by Meng-QQ** [comment]
Thanks Robert. It works as you explained. Cheers!

**Comment by robertmartin8** [comment]
Fixed in v1.3.0
